java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/export/xsl/TransformStyle.java

```diff
@@ -165,6 +173,59 @@ private static void setTextFont(final Element xslCell,
 		}
 	}
 
+	/**
+	 * Set block text content
+	 * 
+	 * @param xslBlock
+	 *            the fo:block element
+	 * @param excelCell
+	 *            the excel cell
+	 */
+	public static void setCellContent(final Element xslBlock,
+			final Cell excelCell) {
+		final Document doc = xslBlock.getOwnerDocument();
+		final RichTextString richStringCellValue = excelCell
+				.getRichStringCellValue();
+		if (richStringCellValue.numFormattingRuns() == 0) {
+			xslBlock.setTextContent(richStringCellValue.getString());
+			return;
+		}
+		final BiFunction<String, String, Element> addSupSubElement = (textValue,
+				attributeValue) -> {
+			final Element inline = XMLDocumentExtensions
+					.createXMLElementWithAttr(doc, FO_INLINE,
+							new XMLAttribute("baseline-shift", attributeValue)); //$NON-NLS-1$
+			inline.setTextContent(textValue);
+			return inline;
+		};
+		if (richStringCellValue instanceof final XSSFRichTextString richtext) {
+			for (int i = 0; i < richtext.numFormattingRuns(); i++) {
+				final int startIndex = richtext.getIndexOfFormattingRun(i);
+				final int endIndex = i + 1 < richtext.numFormattingRuns()
+						? richtext.getIndexOfFormattingRun(i + 1)
+						: richtext.length();
+				final String text = richtext.getString()
+						.substring(startIndex, endIndex);
+				final XSSFFont fontOfFormattingRun = richtext
+						.getFontOfFormattingRun(i);
+				if (fontOfFormattingRun == null) {
+					xslBlock.appendChild(doc.createTextNode(text));
+					continue;
+				}
+
+				final Element formatElement = switch (fontOfFormattingRun
+						.getTypeOffset()) {
+					case Font.SS_SUPER -> addSupSubElement.apply(text, "sub"); //$NON-NLS-1$
```

Comment:
@TruongQuangSB This should apply "super" though

```suggestion
					case Font.SS_SUPER -> addSupSubElement.apply(text, "super"); //$NON-NLS-1$
```